File: 2022/Day6/Challenge1.cs
namespace Day6;

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Challenge1
{
    private Utility utility;

    public Challenge1()
    {
        this.utility = new Utility();
    }

    public static void ChallengeOneSolve()
    {
        string data = Utility.RetrieveData();
        int answer = FindUnique(data);
        Console.WriteLine($"Challenge 1 answer: {answer}");
    }

    public static int FindUnique(string data)
    {
        for (int i = 0; i < data.Length - 3; i++)
        {
            string testSequence = data.Substring(i, 4);
            IEnumerable<char> answer = testSequence.Distinct();
            int testDebug = answer.Count();
            if (answer.Count() == 4)
            {
                return i + 4;
            }
        }
        return -1;
    }
}

File: 2025/src/day1/challenge1.py
from utility.retrieve_data import retrieve_data_from_file_name

def challenge1():
    data = retrieve_data_from_file_name("day1")
    count = 0
    safe_number = 50
    for input_line in data:
        line = input_line.strip()
        if (line[0] == "R"):
            safe_number += int(line[1:]) % 100
        else:
            safe_number -= int(line[1:]) % 100
        if (safe_number > 99):
            safe_number -= 100
        elif (safe_number < 0):
            safe_number += 100
        if (safe_number == 0):
            count += 1
    print(count)


challenge1()